src/main/java/org/spongepowered/api/text/TextFactory.java

```diff
@@ -22,96 +22,73 @@
+ * {@link Texts}.
  */
-interface MessageFactory {
+public interface TextFactory {
 
     /**
-     * Creates a {@link MessageBuilder} with empty text.
+     * Parses the specified JSON text and returns the parsed result.
      *
-     * @return A new message builder with empty text
+     * @param json The valid JSON text
+     * @return The parsed text
+     * @throws IllegalArgumentException If the JSON is invalid
      */
-    MessageBuilder createEmptyBuilder();
+    Text parseJson(String json) throws IllegalArgumentException;
 
     /**
-     * Creates a {@link MessageBuilder.Text} with the specified text.
+     * Parses the specified JSON text leniently and returns the parsed result.
      *
-     * @param text The text for the message
-     * @return A new message builder with the specified text
-     * @see Message.Text
+     * @param json The JSON text
+     * @return The parsed text
+     * @throws IllegalArgumentException If the JSON couldn't be parsed
      */
-    MessageBuilder.Text createTextBuilder(String text);
+    Text parseJsonLenient(String json) throws IllegalArgumentException;
 
     /**
-     * Creates a {@link MessageBuilder.Translatable} with the specified
-     * translation and arguments.
+     * Returns a plain text representation of the {@link Text} without any
+     * formattings.
      *
-     * @param translation The translation to use for the message
-     * @param args The arguments for the translation, can be empty
-     * @return A new message builder with the specified translation and
-     *         arguments
-     * @see Message.Translatable
+     * @param text The text to convert
+     * @return The text converted to plain text
      */
-    MessageBuilder.Translatable createTranslatableBuilder(Translation translation, Object[] args);
+    String toPlain(Text text);
 
     /**
-     * Creates a new {@link MessageBuilder.Selector} with the specified
-     * selector.
+     * Returns a JSON representation of the {@link Text} as used in commands.
      *
-     * @param selector The selector for the message
-     * @return A new message builder with the specified selector
-     * @see Message.Selector
+     * @param text The text to convert
+     * @return The text converted to JSON
      */
-    MessageBuilder.Selector createSelectorBuilder(String selector);
-
-    /**
-     * Creates a new {@link MessageBuilder.Score} with the specified score.
-     *
-     * @param score The score for the message
-     * @return A new message builder with the specified score
-     * @see Message.Score
-     */
-    MessageBuilder.Score createScoreBuilder(Object score); // TODO
-
-    /**
-     * Creates a {@link Message} with the specified plain text. The created
-     * message won't have any formatting or events configured.
-     *
-     * @param text The content of the Message
-     * @return The created {@link Message}
-     */
-    Message.Text createPlain(String text);
+    String toJson(Text text);
```

Comment:
It would be cool if we could choose whether this is minified JSON or pretty JSON.
Minified JSON for config/properties files that do not support multiline entries.
Pretty JSON for configs like HOCON (which is JSON).